specs/01-sharded-archiving.md

```diff
@@ -0,0 +1,288 @@
+    block_height: u64
+    block_hash: [u8; 32]
+    block_proof: [u8;32]
+}
+```
+
+**Table 1: Block Encoding Parameters**
+
+| Parameter              | Description                 | Example Value (Assumed) |
+| ---------------------- | --------------------------- | ----------------------- |
+| `block_size`           | Size of the block in bytes  | 4,000,000 bytes         |
+| `SHARE_SIZE`           | Size of each share in bytes | 4 kbytes                |
+| `source_share`         | Total number of shares      | 100                     |
+| `total_shares`         | Shares after erasure coding | 200                     |
+| `BLOCK_CHUNK_SIZE`     | Size of each chunk in bytes | 32 bytes                |
+| `num_chunks_per_share` | Chunks per share            | 125                     |
+| `total_chunks`         | Total chunks in the block   | 25,000                  |
+
+- Output: Error corrected encoded blocks, and `BlockInfo` to be submitted to the parent shard.
+
+#### 4. Block Info Submission
+
+- Input: `BlockInfo` from a child shard.
+- Any node in the shard (from farmers to full-nodes) are allowed to create the `BlockInfo`
+  transaction that submits the block information into the parent's shard history.
+- The corresponding piece of the block should be stored in the shard and made retrievable via a
+  peer-to-peer network protocol (DSN).
+- When the transaction in the parent
+- Output: Transaction in the parent shard's mempool.
+
+#### 5. Shard block proof in parent
+
+- Input: Transaction with `BlockInfo` from a child shard.
+- The execution of the `BlockInfo` transaction when included in a block of the parent shards commits
+  the block from the child shard to the history of the parent by including it on-chain into the
+  `shard_blocks[]` table.
+- To verify the `BlockInfo`, nodes in the parent shard:
+  - Ensure `block_index` is the immediate block after last committed block number in the parent
+    shard.
+- Output: `BlockInfo` from child shard included in `shard_blocks[]` table of the parent shard.
+
+#### 6. Segment Construction
+
+- Input: Buffered data of size `RAW_RECORD_SIZE`.
+- When the buffer (after encoding) contains enough data to fill a record of `RAW_RECORD_SIZE` bytes,
+  it is archived:
+  1. Split the record into `d = 2^15 =NUM_CHUNKS` chunks of size `SAFE_BYTES`.
+  2. Perform the same operations to extend with parity chunks and commit to the chunks as was done
+     for the block. This is done by creating a Merkle tree of the chunks of a record, and committing
+     to it.
+- Erasure code records column-wise with `extend(column, ERASURE_CODING_RATE)`. This effectively
+  doubles the number of rows and thus, records per segment to `NUM_PIECES` (256).
+- For each row, a proof of the row chunks is created by building a Merkle Proof of the individual
+  chunks in the share, `Cs_i`, along with their proof `ws_i`.
+- For each record, form a `piece = record || record_proof || record_proof`, `(ri∣∣Ci∣∣πi)`
```

Comment:
This says two record proofs, but it is actually different:
https://github.com/nazar-pc/abundance/blob/2760b0ad8f67ae92dc33a22f5f68f161371a80db/subspace/crates/subspace-core-primitives/src/pieces.rs#L1023-L1045

Reply:
This is typo. Thanks for flagging and sharing the reference code.